fulp_modules/features/toys/code/shrimp_plush.dm

```diff
@@ -0,0 +1,37 @@
+/datum/mood_event/shrimp_rice
+	description = "I can't believe a shrimp fried this rice!"
+	mood_change = 3
+	timeout = 2 MINUTES
+
+/obj/item/toy/plush/shrimp/Destroy()
+	. = ..()
+	if(fried_rice)
+		UnregisterSignal(fried_rice, COMSIG_FOOD_EATEN)
+
+/obj/item/toy/plush/shrimp/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	if(proximity_flag && has_fried)
+		return
+
+	if(!istype(user.mind.assigned_role, /datum/job/cook))
+		return
+
+	if(!istype(target, /obj/item/food))
```

Comment:
Doesnt this work on any type of food, not just rice?

Reply:
The idea is that it only works on fried rice foods, hence the check on line 23. Checking the name of the food is a little weird ik, but I couldn't really think of another way to do it.

Reply:
Can someone theoretically just use a pen to rename food to fried rice to make it work?

Reply:
ugh yeah, i'll change it to use the initial value instead